lib/ecto/adapters/sql.ex

```diff
@@ -38,19 +38,15 @@ defmodule Ecto.Adapters.SQL do
       ## Types
 
       @doc false
-      def autogenerate(:id), do: nil
-      def autogenerate(:embed_id), do: Ecto.UUID.autogenerate()
-      def autogenerate(:binary_id), do: Ecto.UUID.autogenerate()
+      def autogenerate(:id),        do: nil
+      def autogenerate(:embed_id),  do: Ecto.UUID.generate()
```

Comment:
But it is wrong in many places.
For SQL databases we want values in embeds to be application-level types, while in regular models to be database level types.
With Mongo, I want the value to always be database level, because there's no JSON encoding in the middle.